proof/proof.go

```diff
@@ -152,6 +151,7 @@ func (p *Proof) DecodeRecords() []tlv.Record {
 		MetaRevealRecord(&p.MetaReveal),
 		AdditionalInputsRecord(&p.AdditionalInputs),
 		ChallengeWitnessRecord(&p.ChallengeWitness),
+		BlockHeightRecord(&p.BlockHeight),
```

Comment:
Should we add this field to one of the unit tests? I would assume that would've caught the "TLV stream must be canonical" error that caused you to move this field to the bottom.

Reply:
In the latest commits, I've added the block height field into a proof encode/decode unit test.